### PR TITLE
Changes on core for 41473

### PR DESCRIPTION
**Pull request by CoMPaTech**
Persistent notifications left in (although changed to lower instead of upper in line with core change for keys)

### CODE REVIEW

File: custom_components/plugwise/binary_sensor.py [hunk 1]
```diff
@@ -184,32 +182,47 @@ async def async_setup_entry_gateway(hass, config_entry, async_add_entities):
...
     def is_on(self):
         """Return true if the binary sensor is on."""
         return self._is_on
 
+    @callback
+    def _async_process_data(self):
+        """Update the entity."""
+        raise NotImplementedError
+
+class GwBinarySensor(SmileSensor, BinarySensorEntity):
```
Comment (bouwew) [inline custom_components/plugwise/binary_sensor.py:223]: Shouldn't this be `...(SmileBinarySensor, ...):`?

**Review by bouwew — CHANGES_REQUESTED** [review]
The enabled_default stuff is missing?

File: custom_components/plugwise/binary_sensor.py (hunk 1 above)
Comment (CoMPaTech) [reply custom_components/plugwise/binary_sensor.py]: Yup

**Comment by CoMPaTech** [comment]
> The enabled_default stuff is missing?

Re-added from sensor

**Comment by CoMPaTech** [comment]
Should be rebased after #140 is merged (due to `enabled_default` changes etc.) - also upstream PR is not reviewed yet

**Review by bouwew — APPROVED** [review]
Ok now it should be complete.

**Comment by CoMPaTech** [comment]
The notification sensor is `default_enabled = False` in `-beta` as we still have the `persistent notifications` :) 

Schermafbeelding 2020-11-16 om 19 55 45

**Review by bouwew — COMMENTED** [review]
Some code is missing:
```
                    if msg_type not in SEVERITIES:
                        msg_type = "other"

                    self._attributes[f"{msg_type.lower()}_msg"].append(msg)
```
Let's keep it close to the code that is/will be in Core.

**Comment by CoMPaTech** [comment]
> Some code is missing:
Added

File: custom_components/plugwise/binary_sensor.py [hunk 2]
```diff
@@ -287,6 +287,9 @@ def _async_process_data(self):
 
             for notify_id, details in notify.items():
                 for msg_type, msg in details.items():
+                    if msg_type not in SEVERITIES:
+                        msg_type = "other"
+
                     self._attributes[msg_type.upper()] = msg
```
Comment (bouwew) [inline custom_components/plugwise/binary_sensor.py:293]: This should say: `self._attributes[f"{msg_type.upper()}_msg"].append(msg)`

Comment (CoMPaTech) [reply custom_components/plugwise/binary_sensor.py]: Lacking copy/paste skills today

Comment (bouwew) [reply custom_components/plugwise/binary_sensor.py]: :)

**Review by bouwew — APPROVED** [review]
Looks like the file is now completed :)

**Pull request merged**